clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -667,15 +667,20 @@ public FileStatus[] listStatus(Path path) throws FileNotFoundException, IOExcept
         OPERATIONS_LOG.trace("list_status({})", path);
         ObjectLocation objectLoc = pathToObjectLocation(path);
         ObjectsApi objectsApi = lfsClient.getObjectsApi();
-        try {
-            ObjectStats objectStat = objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath()).userMetadata(false).presign(false).execute();
-            LakeFSFileStatus fileStatus = convertObjectStatsToFileStatus(objectLoc, objectStat);
-            return new FileStatus[]{fileStatus};
-        } catch (ApiException e) {
-            if (e.getCode() != HttpStatus.SC_NOT_FOUND) {
-                throw new IOException("statObject", e);
+        String ObjectPath = objectLoc.getPath();
+        // If the path is empty, it represents the root of the branch - no need to try to stat
+        if (!ObjectPath.isEmpty()){
```

Comment:
just to be sure, when we list an empty path it will list the files on root?
verify that it works without objects and I think we are good. 
the previous implementation didn't get an exception and fallback to the listing?

Reply:
The previous implementation checked for status code 404 otherwise threw an error, empty path to stat returns a 400 because it's a bad request. I didn't want to ignore bad requests given that it's a specific case.
Yes, when we list an empty path it lists root I was about to write on the issue
I tested it by running val df = spark.read.parquet("lakefs://wild/play/*/*/*.parquet").withColumn("FileName", input_file_name()) and it returned the correct table I also tested val df = spark.read.parquet("lakefs://wild/main/*.parquet").withColumn("FileName", input_file_name()) and it read the lakes.parquet. Hope this answers your question

Reply:
verifying without objects...

Reply:
When running  
`spark.read.parquet("lakefs://wild/main/*.parquet")`:
1. root contains only `lakes.parquet` -> reads the table
2. root contains only `README.md` -> get path not found 
3. root contains nothing -> get path not found